src/utils/result-writer.ts

```diff
@@ -0,0 +1,63 @@
+import ConsoleWriter from './console-writer';
+import { ITaskResult, IConsoleWriter } from '../interfaces';
+
+export default class ResultWriter {
+  results: ITaskResult[];
+  writer: IConsoleWriter;
+
+  constructor(results: ITaskResult[], writer: IConsoleWriter = new ConsoleWriter()) {
+    this.results = results;
+    this.writer = writer;
+  }
+
+  toConsole() {
+    this.writer.line();
+
+    this.results.forEach(result => {
+      result.toConsole(this.writer);
+    });
+  }
+
+  /**
+   * Returns a json object with the project metadata and info of all the tasks that were run.
+   * Example output for a project named foo and its 4 tasks:
+   *  {
+   *    name: 'foo',
+   *    version: 'bar',
+   *    type: 'baz',
+   *    tasks: [
+   *        types: {},
+   *        tests: {},
+   *        angleBracket: {},
+   *        nativeClass: {}
+   *      ],
+   *    dependencies: {}
+   *  }
+   *
+   */
+  toJson() {
+    const resultData = {
+      tasks: [] as object[],
+    };
+
+    this.results.forEach(taskResult => {
```

Comment:
I find this mildly awkward, but can't really articulate why (other than the fact we're explicitly checking for those task types). I'd love to find a more dynamic way to do this, or a way where we can annotate specific classes to output as 'top level' in the result's JSON. No need to change anything here right now, but I just wanted to put it on our radars.